### DISCUSSION TITLE
Multi-repo structure

### DISCUSSION THREAD

**Discussion by josecelano** [Nautilus-Cyberneering/chinese-ideographs#81] [General]
This is a recurrent discussion we have had. How we should split the system into more than one repo. Right now we have two repos:

- Library (this repo): https://github.com/Nautilus-Cyberneering/chinese-ideographs
- Website: https://github.com/Nautilus-Cyberneering/chinese-ideographs-website

For the [POC](https://github.com/Nautilus-Cyberneering/chinese-ideographs/milestone/1) we decided to use only two, but we are considering other approaches like:

- One repo per radical or library (if we consider a radical a different library)
- One parent library repo contains data and code related to the whole system. The code to handle the library could be called the "librarian"

In the current solution (still in an issue branch), the website has a git submodule with the library: https://github.com/Nautilus-Cyberneering/chinese-ideographs-website/tree/issue-1-import-base-images. Through this solution we can import the "library" into the project that is going to consume the media files. In this case, we only need the dvc pointers (data folder in the library repo). In other cases maybe we could only need the "librarian" code, which makes me things if we should split the current repo at least into two repos. That solution perfectly matches with the librarian repo approach. The repo organization could be like this:

- Library: https://github.com/Nautilus-Cyberneering/chinese-ideographs-library (with radicals and drawings and a librarian submodule)
- Website: https://github.com/Nautilus-Cyberneering/chinese-ideographs-website (with a library submodule)
- Librarian: https://github.com/Nautilus-Cyberneering/chinese-ideographs-librarian (domain logic to handle all libraries)

Having the librarian and library in the same repo is not a problem right now, but if we split the library, for sure we are going to need something like that.

**Comment by cgbosse** [comment]
Is this how it would look like? 
![imagen](https://user-images.githubusercontent.com/15330600/138420404-0d115b54-7ec5-4aa4-959f-caf12b0df60c.png)
